src/robot/libraries/BuiltIn.py

```diff
@@ -1662,26 +1662,58 @@ def run_keyword_and_expect_error(self, expected_error, name, *args):
         return unic(error)
 
     @run_keyword_variant(resolve=2)
-    def repeat_keyword(self, times, name, *args):
+    def repeat_keyword(self, repeat, name, *args):
         """Executes the specified keyword multiple times.
 
         ``name`` and ``args`` define the keyword that is executed
-        similarly as with `Run Keyword`, and ``times`` specifies how many
-        the keyword should be executed. ``times`` can be given as an
+        similarly as with `Run Keyword`.
+
+        If ``repeat`` is given as count, it specifies how many the keyword
+        should be executed. ``repeat`` can be given as an
         integer or as a string that can be converted to an integer. If it is
         a string, it can have postfix ``times`` or ``x`` (case and space
         insensitive) to make the expression more explicit.
 
-        If ``times`` is zero or negative, the keyword is not executed at
+        If ``repeat`` is given as timeout, it must be in Robot Framework's
+        time format (e.g. ``1 minute``, ``2 min 3 s``) except a number like
+        42 is interpreted as count, so that a floating point number like
+        4.2 would raise exception.
+        You can use ``x`` as millisecond only with ``repeat`` including other
+        string that can be interpreted as timeout such as ``1 s 10 x``. Time
+        string only with ``x`` is interpreted as count.
+
+        If ``repeat`` is zero or negative, the keyword is not executed at
         all. This keyword fails immediately if any of the execution
         rounds fails.
 
         Examples:
         | Repeat Keyword | 5 times | Go to Previous Page |
         | Repeat Keyword | ${var}  | Some Keyword | arg1 | arg2 |
+        | Repeat Keyword | 2 mins  | Some Keyword | arg1 | arg2 |
         """
-        times = self._get_times_to_repeat(times)
-        self._run_keywords(self._yield_repeated_keywords(times, name, args))
+        maxtime = count = None
+        try:
+            _ = float(repeat)
+        except:
+            isnumeric = False
+        else:
+            isnumeric = True
+        if not isnumeric:
+            try:
+                timeout = timestr_to_secs(repeat)
+                maxtime = time.time() + timeout
+            except:
+                pass
+        try:
+            count = self._get_times_to_repeat(repeat)
+        except Exception as err:
+            if not maxtime:
+                raise err
+        if count:
+            kwiter = self._yield_repeated_keywords(count, name, args)
+        else:
+            kwiter = self._yield_repeated_keywords_until(maxtime, name, args)
+        self._run_keywords(kwiter)
```

Comment:
@pekkaklarck, Thanks for your suggestion.
I implemented this just like your code for the first time. The code was becoming complicated for the backward compatibility issues.
With your code, the test case "Invalid Times" in acceptance test (repeat_keyword.robot) fails.
- Invalid Times 1
  : 1.3 does not make failure. It is interpreted into 1.3 sec
- Invalid Times 2
  : FAIL message is different, because the exception is raised by timestr_to_secs().

The issues cannot be a problem with the modification of the existing test case "Invalid Times".
Is it OK with you?